tgui/packages/tgui/interfaces/Manufacturer/type.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+type ManufacturerData = {
+  all_categories:[string];
+
+  card_owner:string;
+  fabricator_name:string;
+
+  available_blueprints:{ [key:string] : Manufacturable[]};
+  downloaded_blueprints:{ [key:string] : Manufacturable[]};
+  drive_recipe_blueprints:{ [key:string] : Manufacturable[]};
+  hidden_blueprints:{ [key:string] : Manufacturable[]};
+
+  resources:{ [key: string] : number };
+  wires:{ [key:string] : string};
+
+  delete_allowed:boolean;
```

Comment:
ya and oh ok. any reason to use boolean over BooleanLike then?

Reply:
Yes, it's still entirely valid in a whole bunch of places. You could have a `useLocalState` that stores a `boolean`, for example. As that's not coming from `ui_data` it will be an actual `boolean` instead of a `BooleanLike`. Things like the response of a `.filter` callback will still be `boolean`. `BooleanLike` doesn't even stop you doing the thing I said above, or warn about it, but at least if you look at the type you know you need to handle differently (common one would be to double-negate for the sake of getting it into an actual boolean).

Reply:
👍 nice! thank you for the explanation on that, that helps.